apsuite/optics_analysis/misalign_analysis/base.py

```diff
@@ -166,27 +166,40 @@ def __force_init(self, elems, sects, dtypes):
         # gen buttons
         self._buttons = self.__generate_buttons()
 
+    def __search_button_in_default_base(self, button):
+        for b in globals()["_DEFAULT_BUTTONS"]:
+            if (
+                (button.dtype == b.dtype)
+                and (button.indices == b.indices)
+                and (button.fantasy_name == b.fantasy_name)
+            ):
+                return True, b
+        return False, None
+
     def __generate_buttons(self):
         all_buttons = []
         for dtype in self._dtypes:
```

Comment:
this structure is too nested. Consider using `itertools.product` to make a simple for loop.
```python3
from itertools import product
for i, j, k in product([1, 2], ['a', 'b'], [3, 4]):
    print(i, j, k)
```
would result in:
```
1 a 3
1 a 4
1 b 3
1 b 4
2 a 3
2 a 4
2 b 3
2 b 4
```